programs/autocrat_v0/src/lib.rs

```diff
@@ -151,147 +152,84 @@ pub mod autocrat_v0 {
-
-        require!(
-            openbook_twap_pass_market.twap_oracle.initial_slot + 50 >= clock.slot,
-            AutocratError::TWAPMarketTooOld
-        );
-        require!(
-            openbook_twap_fail_market.twap_oracle.initial_slot + 50 >= clock.slot,
-            AutocratError::TWAPMarketTooOld
-        );
-        require_eq!(
-            openbook_twap_pass_market
-                .twap_oracle
-                .max_observation_change_per_update_lots,
-            dao.max_observation_change_per_update_lots,
-            AutocratError::TWAPOracleWrongChangeLots
-        );
-        require_eq!(
-            openbook_twap_fail_market
-                .twap_oracle
-                .max_observation_change_per_update_lots,
-            dao.max_observation_change_per_update_lots,
-            AutocratError::TWAPOracleWrongChangeLots
-        );
-        require!(
-            openbook_twap_pass_market.twap_oracle.expected_value == dao.twap_expected_value,
-            AutocratError::TWAPMarketInvalidExpectedValue
-        );
-        require!(
-            openbook_twap_fail_market.twap_oracle.expected_value == dao.twap_expected_value,
-            AutocratError::TWAPMarketInvalidExpectedValue
-        );
-
-        let proposal = &mut ctx.accounts.proposal;
-
-        proposal.number = dao.proposal_count;
-        dao.proposal_count += 1;
-
-        // least signficant 32 bits of nonce are proposal number
-        // most significant bit of nonce is 0 for base (META) and 1 for quote (USDC)
-        require!(
-            ctx.accounts.base_vault.nonce == proposal.number as u64,
-            AutocratError::InvalidVaultNonce
-        );
-        require!(
-            ctx.accounts.quote_vault.nonce == (proposal.number as u64 | (1 << 63)),
-            AutocratError::InvalidVaultNonce
-        );
+        for market in [&pass_market, &fail_market] {
+            // The market expires a minimum of 7 days after the end of a 3 day proposal.
+            // Make sure to do final TWAP crank after the proposal period has ended
+            // and before the market expires, or else! Allows for rent retrieval from openbook
+            require!(
+                market.time_expiry > clock.unix_timestamp as i64 + TEN_DAYS_IN_SECONDS,
+                AutocratError::InvalidMarket
+            );
+
+            require!(
+                market.taker_fee == dao.market_taker_fee,
+                AutocratError::InvalidMarket
+            );
+
+            require!(market.maker_fee == 0, AutocratError::InvalidMarket);
+
+            require!(
+                market.base_lot_size == dao.base_lot_size,
+                AutocratError::InvalidMarket
+            );
+
+            require!(
+                market.quote_lot_size == 100, // you can quote in increments of a hundredth of a penny
+                AutocratError::InvalidMarket
+            );
+
+            require!(
+                pass_market.collect_fee_admin == dao.treasury,
```

Comment:
chatgpt spotted this one. It still has `pass_market` and not `market`

Reply:
good catch! fixed